src/main/java/pl/bartoszmech/domain/offer/OfferService.java

```diff
@@ -0,0 +1,49 @@
+package pl.bartoszmech.domain.offer;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.stereotype.Service;
+import pl.bartoszmech.domain.offer.dto.OfferApiResponse;
+import pl.bartoszmech.domain.offer.dto.OfferResponse;
+
+import java.util.List;
+import java.util.Objects;
+
+@Log4j2
+@AllArgsConstructor
+@Service
+public class OfferService {
+    OfferFetcher fetcher;
```

Comment:
add private to your dependencies or add @RequiredArgsConstructor and make them final